samples/Foundatio.HostingSample/Program.cs

```diff
@@ -90,7 +92,14 @@ public static IHostBuilder CreateHostBuilder(string[] args)
                     s.AddCronJob<EveryMinuteJob>("* * * * *");
 
                 if (evenMinutes)
-                    s.AddCronJob<EvenMinutesJob>("*/2 * * * *");
+                    s.AddCronJob("*/2 * * * *", async sp =>
+                    {
+                        var logger = sp.GetRequiredService<ILogger<Program>>();
+                        if (logger.IsEnabled(LogLevel.Information))
+                            logger.LogInformation("EvenMinuteJob Run Thread={ManagedThreadId}", Thread.CurrentThread.ManagedThreadId);
```

Comment:
Wonder if we could inject a logger based on the job name string as a parameter. Then again you have the provider

Reply:
No, not all jobs need a logger.